### ISSUE TITLE
Use sizes on <Source> to generate srcSet

### ISSUE THREAD

**Issue by dbow** [imgix/react-imgix#243]
**Before you submit:**

- [x] Please search through the existing issues (both open AND closed) to see if your issue has been discussed before. Github issue search can be used for this: https://github.com/imgix/react-imgix/issues?utf8=%E2%9C%93&q=is%3Aissue
- [x] Please ensure the problem has been isolated and reduced. This link explains more: http://css-tricks.com/6263-reduced-test-cases/

**Describe the bug**

It looks like the `<Source>` component ignores the `sizes=` attribute and just generates a `srcSet` with different resolutions. I think it's [this line](https://github.com/imgix/react-imgix/blob/master/src/react-imgix.jsx#L96):
```
if (fixedSize || type === "source") {
```
But I _think_ `<source>` should be able to support fluid width images the same as `<img>`? Is there a reason that `<Source>` components are grouped in with fixed size images?

**To Reproduce**
```
<Picture>
  <Source
    src={desktopSrc}
    sizes="100vw"
    htmlAttributes={{ media: "(min-width: 768px)" }}
  />
  <Source
    src={mobileSrc}
    sizes="100vw"
    htmlAttributes={{ media: "(min-width: 320px)" }}
  />
  <Imgix src={mobileSrc} sizes="100vw" />
</Picture>
```

**Expected behaviour**

I'd expect a `<Source>` to generate its `srcSet` based on `sizes=` similar to the `<Imgix>` component - otherwise art-directed images end up much larger than they need to be.

**Comment by frederickfogerty** [comment]
Yep, you're dead right @dbow. I'll add this to our backlog to fix (and it may take a while to fix), but would also accept a PR if you'd like to fix this yourself!

**Comment by JamieLottering** [comment]
Any updates on this?

**Comment by sherwinski** [comment]
Hey @JamieLottering, I'm looking to work on this sometime in the next week or so.

**Comment by JamieLottering** [comment]
> Hey @JamieLottering, I'm looking to work on this sometime in the next week or so.

Thank you!

**Comment by sherwinski** [comment]
hey @JamieLottering, these changes are now available as part of [v8.6.3](https://github.com/imgix/react-imgix/releases/tag/v8.6.3) ✌️

**Comment by JamieLottering** [comment]
Thank you, much appreciated!! 🙇
